prow/test/integration/test/webhook_server_test.go

```diff
@@ -55,11 +67,64 @@ func TestWebhookServerValidateCluster(t *testing.T) {
 	//prowjob should be created successfully
 	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidProwJobPath).Run()
 	if err != nil {
-		t.Errorf("prowjob was not properly validated")
+		t.Errorf("prowjob was not properly validated %v", err)
 	}
 	//cleanup prowjob that was created
 	err = cleanUpCluster(validProwJobID)
 	if err != nil {
 		t.Errorf("could not delete prowjob %s", validProwJobID)
 	}
 }
+
+func TestWebhookServerMutateProwjob(t *testing.T) {
+	validBareProwJobPath := "./testdata/valid_bare_prowjob.yaml"
+	absValidBareProwJobPath, err := filepath.Abs(validBareProwJobPath)
+	var mutatedProwJob v1.ProwJob
+	if err != nil {
+		t.Fatalf("could not find absolute file path")
+	}
+	//prowjob should be created successfully
+	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidBareProwJobPath).Run()
+	if err != nil {
+		t.Errorf("prowjob was not properly validated %v", err)
+	}
+
+	key := types.NamespacedName{
+		Namespace: "default",
+		Name:      validBareProwJobID,
+	}
+	cfg := config.GetConfigOrDie()
+	if err != nil {
+		t.Fatalf("Error getting kubeconfig")
+	}
+	client, err := ctrlruntimeclient.New(cfg, ctrlruntimeclient.Options{})
+	if err != nil {
+		t.Fatalf("Could not create writer client %v", err)
+	}
```

Comment:
```suggestion
	clusterContext := getClusterContext()
	t.Logf("Creating client for cluster: %s", clusterContext)
	kubeClient, err := NewClients("", clusterContext)
	if err != nil {
		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
	}
```

please use the shared method for integration test to make sure it's not targeting the cluster that your `kubectl` points to at the moment, which is very dangerous for someone who debugs this test, as it will polute prod services